MvvmCross/Core/Core/Platform/MvxSimplePropertyDictionaryExtensionMethods.cs

```diff
@@ -148,14 +149,25 @@ public static string GetPropertyValueAsString(this object input, PropertyInfo pr
             try
             {
                 var value = propertyInfo.GetValue(input, new object[] { });
-
-                return value?.ToString();
+                return value?.ToStringInvariant();
             }
             catch (Exception suspectedMethodAccessException)
             {
                 throw suspectedMethodAccessException.MvxWrap(
                     "Problem accessing object - most likely this is caused by an anonymous object being generated as Internal - please see http://stackoverflow.com/questions/8273399/anonymous-types-and-get-accessors-on-wp7-1");
             }
         }
+
+        private static string ToStringInvariant(this object value)
```

Comment:
``Convert.ToString(value, CultureInfo.InvariantCulture)`` causes precision dropping.
So DateTime, double,  float types need to use round-trip conversion in ```ToStringInvariant```.

example
```c#
var str = Convert.ToString(double.MaxValue,  CultureInfo.InvariantCulture);
var result = double.TryParse(str, out var value); // result is false because str becomed value with low precision 
```